test/cloudtest/test_crash.py

```diff
@@ -97,13 +99,50 @@ def test_crash_storaged(mz: MaterializeApplication) -> None:
 
 
 def test_crash_environmentd(mz: MaterializeApplication) -> None:
+    def epoch(p: V1StatefulSet) -> int:
+        epoch_label = "environmentd.materialize.cloud/epoch"
+        assert p.metadata is not None
+        assert p.metadata.labels is not None
+        return int(p.metadata.labels[epoch_label])
+
+    def restarts(p: V1Pod) -> int:
+        assert p.status is not None
+        assert p.status.container_statuses is not None
+        return p.status.container_statuses[0].restart_count
+
+    def get_replica() -> Tuple[V1Pod, V1StatefulSet]:
+        """Find the stateful set for the replica of the default cluster"""
+        compute_pod_name = f"compute-cluster-u1-replica-1-0"
+        ss_name = f"compute-cluster-u1-replica-1"
+        compute_pod = mz.environmentd.api().read_namespaced_pod(
+            compute_pod_name, mz.environmentd.namespace()
+        )
+        for ss in (
+            mz.environmentd.apps_api().list_stateful_set_for_all_namespaces().items
+        ):
+            assert ss.metadata is not None
+            if ss.metadata.name == ss_name:
+                return (compute_pod, ss)
+        assert False
+
     populate(mz, 2)
+
+    before = get_replica()
+
     try:
         mz.environmentd.sql("SELECT mz_internal.mz_panic('forced panic')")
     except InterfaceError:
         pass
     validate(mz, 2)
 
+    after = get_replica()
+
+    # A environmentd crash must increment the epoch
+    assert epoch(before[1]) + 1 == epoch(after[1])
```

Comment:
Check for less than rather than equality. The epoch is not guaranteed to be incremented by 1.

Reply:
gone now